lib/web_ui/lib/src/engine/html/shader_mask.dart

```
@@ -0,0 +1,376 @@
+      // Porter duff source * destination but preserves alpha.
+      svgFilter = _modulateImageToSvg(imageUrl);
+      break;
+    case ui.BlendMode.overlay:
+      // Since overlay is the same as hard-light by swapping layers,
+      // pass hard-light blend function.
+      svgFilter = _blendImageToSvg(imageUrl, 'hard-light', swapLayers: true);
+      break;
+    // Several of the filters below (although supported) do not render the
+    // same (close but not exact) as native flutter when used as blend mode
+    // for a background-image with a background color. They only look
+    // identical when feBlend is used within an svg filter definition.
+    //
+    // Saturation filter uses destination when source is transparent.
+    // cMax = math.max(r, math.max(b, g));
+    // cMin = math.min(r, math.min(b, g));
+    // delta = cMax - cMin;
+    // lightness = (cMax + cMin) / 2.0;
+    // saturation = delta / (1.0 - (2 * lightness - 1.0).abs());
+    case ui.BlendMode.saturation:
+    case ui.BlendMode.colorDodge:
+    case ui.BlendMode.colorBurn:
+    case ui.BlendMode.hue:
+    case ui.BlendMode.color:
+    case ui.BlendMode.luminosity:
+    case ui.BlendMode.multiply:
+    case ui.BlendMode.screen:
+    case ui.BlendMode.overlay:
+    case ui.BlendMode.darken:
+    case ui.BlendMode.lighten:
+    case ui.BlendMode.colorDodge:
+    case ui.BlendMode.colorBurn:
+    case ui.BlendMode.hardLight:
+    case ui.BlendMode.softLight:
+    case ui.BlendMode.difference:
+    case ui.BlendMode.exclusion:
+      svgFilter = _blendImageToSvg(imageUrl, _stringForBlendMode(blendMode));
+      break;
+    case ui.BlendMode.src:
+    case ui.BlendMode.dst:
+    case ui.BlendMode.dstATop:
+    case ui.BlendMode.dstIn:
+    case ui.BlendMode.dstOut:
+    case ui.BlendMode.dstOver:
+    case ui.BlendMode.clear:
+    case ui.BlendMode.srcOver:
+      assert(
+          false,
+          'Invalid svg filter request for blend-mode '
+          '$blendMode');
```

Instead of `assert` followed by `break` let's just `throw UnsupportedError`.